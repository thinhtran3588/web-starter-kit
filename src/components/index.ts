export * from './AdminLayout';
export * from './AdminFooter';
export * from './Footer';
export * from './Header';
export * from './Icon';
export * from './Layout';
export * from './Sidebar';
export * from './SidebarMenu';
export * from './FormHeader';
export * from './Field';
export * from './FormFilter';
export * from './FormInput';
export * from './FormSearch';
export * from './Table';
export * from './LanguageSelection';
export * from './Link';
export * from './AuthLayout';
export * from './Head';
export * from './Button';
export * from './Grid';
export * from './Typography';
