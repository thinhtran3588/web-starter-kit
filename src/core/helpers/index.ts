export * from './format_with_lookup';
export * from './format_date';
export * from './format_date_time';
export * from './show_notification';
export * from './catch_error';
export * from './handle_error';
export * from './app_error';
export * from './sanitize_form_data';
export * from './get_error_message';
export * from './sleep';
