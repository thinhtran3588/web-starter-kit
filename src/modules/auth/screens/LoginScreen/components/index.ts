export * from './EmailLogin';
export * from './PhoneNoLogin';
