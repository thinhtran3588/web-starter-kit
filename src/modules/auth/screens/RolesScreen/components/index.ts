export * from './Detail';
export * from './PermissionsTable';
