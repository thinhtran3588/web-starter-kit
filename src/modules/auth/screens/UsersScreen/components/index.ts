export * from './Detail';
export * from './PermissionsTable';
export * from './ChangePassword';
