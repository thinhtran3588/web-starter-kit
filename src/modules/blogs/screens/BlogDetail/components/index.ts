export * from './Content';
export * from './SideMenu';
