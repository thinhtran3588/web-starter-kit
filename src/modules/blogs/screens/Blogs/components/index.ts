export * from './Categories';
export * from './BlogsList';
