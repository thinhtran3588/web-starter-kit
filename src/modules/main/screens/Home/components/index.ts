export * from './CoverImage';
export * from './Features';
export * from './Quote';
export * from './Help';
