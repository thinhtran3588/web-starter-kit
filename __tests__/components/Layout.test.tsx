import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Layout } from '@app/components';
import { TestBaseComponent } from '@test/helpers/TestBaseComponent';

describe('components/Layout', () => {
  beforeEach(() => {});
  it('renders successfully', async () => {
    const { baseElement } = render(
      <TestBaseComponent>
        <Layout />
      </TestBaseComponent>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('renders successfully with title', async () => {
    const { baseElement } = render(
      <TestBaseComponent>
        <Layout title='Title' />
      </TestBaseComponent>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('opens sidebar when clicking Menu icon', async () => {
    const { getByTestId } = render(
      <TestBaseComponent>
        <Layout title='Title' />
      </TestBaseComponent>,
    );
    fireEvent.click(getByTestId('menu-icon'));

    expect(getByTestId('sidebar')).toBeDefined();
    expect(getByTestId('sidebar').getAttribute('aria-hidden')).not.toBe('true');
  });

  it('closes sidebar when clicking sidebar backdrop', async () => {
    const { getByTestId } = render(
      <TestBaseComponent>
        <Layout title='Title' />
      </TestBaseComponent>,
    );
    fireEvent.click(getByTestId('menu-icon'));
    const backdrop = getByTestId('sidebar').querySelector('.MuiBackdrop-root');
    if (!backdrop) {
      expect(true).toEqual(false);
      return;
    }
    fireEvent.click(backdrop);

    expect(getByTestId('sidebar').getAttribute('aria-hidden')).toBe('true');
  });
});
