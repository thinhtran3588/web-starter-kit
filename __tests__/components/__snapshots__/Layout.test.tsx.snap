// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout renders successfully 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-root-3 MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <next-link
          href="/"
        >
          <a
            class=""
            href="/"
            title="Home"
          >
            <img
              alt="Logo"
              class="makeStyles-logo-5"
              src="/static/images/logo.svg"
            />
          </a>
        </next-link>
        <div
          class="makeStyles-flexGrow-4"
        />
        <div>
          <button
            aria-label="common.changeLanguage"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <div
                class="MuiAvatar-root makeStyles-icon-52 MuiAvatar-colorDefault"
              >
                EN
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <button
          aria-label="notification"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              class="MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                />
              </svg>
              <span
                class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-dot"
              />
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          data-testid="menu-icon"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <div
      class="makeStyles-container-1"
    >
      <main
        class="makeStyles-main-2"
      />
      <footer
        class="makeStyles-root-109"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-lg-2"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-lg-8"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-3"
              >
                <h3
                  class="makeStyles-company-110"
                >
                  Company
                </h3>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    About us
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Contact
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Products
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Blogs
                  </a>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-3"
              >
                <h3
                  class="makeStyles-company-110"
                >
                  Legal
                </h3>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    FAQ
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Premium support
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Sponsorships
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Custom development
                  </a>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
              >
                <h3
                  class="makeStyles-company-110"
                >
                  Social medias
                </h3>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-socialLink-111 MuiTypography-colorInherit"
                  >
                    <img
                      class="makeStyles-image-112"
                      src="/static/images/facebook.png"
                    />
                  </a>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-socialLink-111 MuiTypography-colorInherit"
                  >
                    <img
                      class="makeStyles-image-112"
                      src="/static/images/dribbble.png"
                    />
                  </a>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-socialLink-111 MuiTypography-colorInherit"
                  >
                    <img
                      class="makeStyles-image-112"
                      src="/static/images/google-plus.png"
                    />
                  </a>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-socialLink-111 MuiTypography-colorInherit"
                  >
                    <img
                      class="makeStyles-image-112"
                      src="/static/images/linkedin.png"
                    />
                  </a>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-socialLink-111 MuiTypography-colorInherit"
                  >
                    <img
                      class="makeStyles-image-112"
                      src="/static/images/pinterest.png"
                    />
                  </a>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-signature-113"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                © 
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  href="/"
                >
                  Thinh Tran
                </a>
                 
                2019
              </p>
              <span
                class="MuiTypography-root MuiTypography-caption"
              >
                common.footer.createdWithLove
              </span>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-lg-2"
          />
        </div>
      </footer>
    </div>
  </div>
  <div
    aria-hidden="true"
    class="MuiPopover-root"
    id="simple-menu"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
  >
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiMenu-paper MuiPaper-elevation8 MuiPopover-paper MuiPaper-rounded"
      style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiMenu-list MuiList-padding"
        role="menu"
        tabindex="-1"
      >
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="0"
        >
          English
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="-1"
        >
          Tiếng Việt
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`components/Layout renders successfully with title 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-root-284 MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <next-link
          href="/"
        >
          <a
            class=""
            href="/"
            title="Home"
          >
            <img
              alt="Logo"
              class="makeStyles-logo-286"
              src="/static/images/logo.svg"
            />
          </a>
        </next-link>
        <div
          class="makeStyles-flexGrow-285"
        />
        <div>
          <button
            aria-label="common.changeLanguage"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <div
                class="MuiAvatar-root makeStyles-icon-333 MuiAvatar-colorDefault"
              >
                EN
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <button
          aria-label="notification"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              class="MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                />
              </svg>
              <span
                class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-dot"
              />
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          data-testid="menu-icon"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <div
      class="makeStyles-container-282"
    >
      <main
        class="makeStyles-main-283"
      />
      <footer
        class="makeStyles-root-390"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-lg-2"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-lg-8"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-3"
              >
                <h3
                  class="makeStyles-company-391"
                >
                  Company
                </h3>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    About us
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Contact
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Products
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Blogs
                  </a>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-3"
              >
                <h3
                  class="makeStyles-company-391"
                >
                  Legal
                </h3>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    FAQ
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Premium support
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Sponsorships
                  </a>
                </div>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  >
                    Custom development
                  </a>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
              >
                <h3
                  class="makeStyles-company-391"
                >
                  Social medias
                </h3>
                <div>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-socialLink-392 MuiTypography-colorInherit"
                  >
                    <img
                      class="makeStyles-image-393"
                      src="/static/images/facebook.png"
                    />
                  </a>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-socialLink-392 MuiTypography-colorInherit"
                  >
                    <img
                      class="makeStyles-image-393"
                      src="/static/images/dribbble.png"
                    />
                  </a>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-socialLink-392 MuiTypography-colorInherit"
                  >
                    <img
                      class="makeStyles-image-393"
                      src="/static/images/google-plus.png"
                    />
                  </a>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-socialLink-392 MuiTypography-colorInherit"
                  >
                    <img
                      class="makeStyles-image-393"
                      src="/static/images/linkedin.png"
                    />
                  </a>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-socialLink-392 MuiTypography-colorInherit"
                  >
                    <img
                      class="makeStyles-image-393"
                      src="/static/images/pinterest.png"
                    />
                  </a>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-signature-394"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                © 
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  href="/"
                >
                  Thinh Tran
                </a>
                 
                2019
              </p>
              <span
                class="MuiTypography-root MuiTypography-caption"
              >
                common.footer.createdWithLove
              </span>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-lg-2"
          />
        </div>
      </footer>
    </div>
  </div>
  <div
    aria-hidden="true"
    class="MuiPopover-root"
    id="simple-menu"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
  >
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiMenu-paper MuiPaper-elevation8 MuiPopover-paper MuiPaper-rounded"
      style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiMenu-list MuiList-padding"
        role="menu"
        tabindex="-1"
      >
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="0"
        >
          English
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="-1"
        >
          Tiếng Việt
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
